src/geminim.nim

```diff
@@ -30,6 +30,55 @@ proc getUserDir(path: string): (string, string) =
 
   result = (path[2..<i], path[i..^1])
 
+proc receiveFile(server: Server, client: AsyncSocket, path: string): Future[Response] {.async.} =
+  let params = path.split(";")
+  if params.len < 2:
+    return response(StatusMalformedRequest, "")
```

Comment:
The response function could be rewritten as `proc response*(code: RespStatus, meta = ""): Response` so this part here would be just `return response(StatusMalformedRequest)